packages/reactive-element/src/css-tag.ts

```diff
@@ -15,21 +15,36 @@ export const supportsAdoptingStyleSheets =
   'adoptedStyleSheets' in Document.prototype &&
   'replace' in CSSStyleSheet.prototype;
 
+/**
+ * A CSSResult or native CSSStyleSheet.
+ *
+ * In browsers that support constructible CSS style sheets, CSSStyleSheet
+ * object can be used for styling along side CSSResult from the `css`
+ * template tag.
+ */
 export type CSSResultOrNative = CSSResult | CSSStyleSheet;
 
-export type CSSResultFlatArray = CSSResultOrNative[];
-
-export type CSSResultArray = Array<CSSResultOrNative | CSSResultArray>;
-
-export type CSSResultGroup = CSSResultOrNative | CSSResultArray;
+/**
+ * A single CSSResult, CSSStyleSheet, or an array or nested arrays of those.
+ */
+export type CSSResultGroup =
+  | CSSResultOrNative
+  | Array<CSSResultOrNative | CSSResultGroup>;
```

Comment:
I think this is the same:
```suggestion
export type CSSResultGroup = CSSResultOrNative | Array<CSSResultGroup>;
```